src/services/embeddings_service.py

```
@@ -0,0 +1,278 @@
+
+            # Generate embeddings
+            response = await self._client.embeddings.create(input=truncated_texts, model=model)
+
+            # Extract embeddings
+            embeddings = [item.embedding for item in response.data]
+
+            # Return single embedding or list based on input
+            return embeddings[0] if not is_batch else embeddings
+
+        except Exception as e:
+            logger.error(f"OpenAI embedding generation failed: {e}")
```

Pylint (3.3.7)

[warning] Use lazy % formatting in logging functions

(W1203)